tests/Database/DatabaseEloquentBelongsToManyTest.php

```diff
@@ -116,7 +116,7 @@ public function testModelsAreProperlyMatchedToParents()
         $this->assertEquals(2, $models[1]->foo[0]->pivot->user_id);
         $this->assertEquals(2, $models[1]->foo[1]->pivot->user_id);
         $this->assertEquals(2, count($models[1]->foo));
-        $this->assertEquals(0, count($models[2]->foo));
+        $this->assertEquals(0, count((array) $models[2]->foo));
```

Comment:
Put another way; it would seem, according to how this test is written, that you cannot *safely* use `count()` on a model that is similar in principal to `$models[2]->foo` in PHP7.2

So just because the test passes doesnt mean we are solving the underlying issue?